B039 considers `frozenset` to be mutable

```
$ cat foo.py 
from contextvars import ContextVar

foo = ContextVar('foo', default=frozenset[str]())
$ ruff check --select B039 foo.py
foo.py:3:33: B039 Do not use mutable data structures for `ContextVar` defaults
  |
1 | from contextvars import ContextVar
2 | 
3 | foo = ContextVar('foo', default=frozenset[str]())
  |                                 ^^^^^^^^^^^^^^^^ B039
  |
  = help: Replace with `None`; initialize with `.set()``

Found 1 error.
$ ruff --version                 
ruff 0.8.0
```

The error is also silenced if the context var is an *annotated* mutable type:

```
$ cat foo.py
from contextvars import ContextVar

foo = ContextVar[set[str]]('foo', default=set())  # No error
bar = ContextVar('bar', default=set())
$ ruff check --select B039 foo.py
foo.py:4:33: B039 Do not use mutable data structures for `ContextVar` defaults
  |
3 | foo = ContextVar[set[str]]('foo', default=set())  # No error
4 | bar = ContextVar('bar', default=set())
  |                                 ^^^^^ B039
  |
  = help: Replace with `None`; initialize with `.set()``

Found 1 error.
```

--- a/crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B039.py
+++ b/crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B039.py
@@ -14,6 +14,8 @@ ContextVar("cv", default=frozenset())
 ContextVar("cv", default=MappingProxyType({}))
 ContextVar("cv", default=re.compile("foo"))
 ContextVar("cv", default=float(1))
+ContextVar("cv", default=frozenset[str]())
+ContextVar[frozenset[str]]("cv", default=frozenset[str]())
 
 # Bad
 ContextVar("cv", default=[])
@@ -25,6 +27,8 @@ ContextVar("cv", default=[char for char in "foo"])
 ContextVar("cv", default={char for char in "foo"})
 ContextVar("cv", default={char: idx for idx, char in enumerate("foo")})
 ContextVar("cv", default=collections.deque())
+ContextVar("cv", default=set[str]())
+ContextVar[set[str]]("cv", default=set[str]())
 
 def bar() -> list[int]:
     return [1, 2, 3]

--- a/crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_contextvar_default.rs
+++ b/crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_contextvar_default.rs
@@ -1,5 +1,6 @@
 use ruff_diagnostics::{Diagnostic, Violation};
 use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::helpers::map_subscript;
 use ruff_python_ast::name::QualifiedName;
 use ruff_python_ast::{self as ast, Expr};
 use ruff_python_semantic::analyze::typing::{is_immutable_func, is_mutable_expr, is_mutable_func};
@@ -92,11 +93,11 @@ pub(crate) fn mutable_contextvar_default(checker: &mut Checker, call: &ast::Expr
         || matches!(
             default,
             Expr::Call(ast::ExprCall { func, .. })
-                if !is_mutable_func(func, checker.semantic())
-                    && !is_immutable_func(func, checker.semantic(), &extend_immutable_calls)))
+                if !is_mutable_func(map_subscript(func), checker.semantic())
+                    && !is_immutable_func(map_subscript(func), checker.semantic(), &extend_immutable_calls)))
         && checker
             .semantic()
-            .resolve_qualified_name(&call.func)
+            .resolve_qualified_name(map_subscript(&call.func))
             .is_some_and(|qualified_name| {
                 matches!(qualified_name.segments(), ["contextvars", "ContextVar"])
             })

--- a/crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_contextvar_default.rs
+++ b/crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_contextvar_default.rs
@@ -93,8 +93,8 @@ pub(crate) fn mutable_contextvar_default(checker: &mut Checker, call: &ast::Expr
         || matches!(
             default,
             Expr::Call(ast::ExprCall { func, .. })
-                if !is_mutable_func(map_subscript(func), checker.semantic())
-                    && !is_immutable_func(map_subscript(func), checker.semantic(), &extend_immutable_calls)))
+                if !is_mutable_func(func, checker.semantic())
+                    && !is_immutable_func(func, checker.semantic(), &extend_immutable_calls)))
         && checker
             .semantic()
             .resolve_qualified_name(map_subscript(&call.func))

--- a/crates/ruff_python_semantic/src/analyze/typing.rs
+++ b/crates/ruff_python_semantic/src/analyze/typing.rs
@@ -279,7 +279,7 @@ pub fn is_immutable_func(
     extend_immutable_calls: &[QualifiedName],
 ) -> bool {
     semantic
-        .resolve_qualified_name(func)
+        .resolve_qualified_name(map_subscript(func))
         .is_some_and(|qualified_name| {
             is_immutable_return_type(qualified_name.segments())
                 || extend_immutable_calls
@@ -306,7 +306,7 @@ pub fn is_mutable_expr(expr: &Expr, semantic: &SemanticModel) -> bool {
         | Expr::ListComp(_)
         | Expr::DictComp(_)
         | Expr::SetComp(_) => true,
-        Expr::Call(ast::ExprCall { func, .. }) => is_mutable_func(func, semantic),
+        Expr::Call(ast::ExprCall { func, .. }) => is_mutable_func(map_subscript(func), semantic),
         _ => false,
     }
 }

